artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ActiveMQServerImpl.java

```diff
@@ -3512,7 +3513,14 @@ public Queue createQueue(final QueueConfiguration queueConfiguration, boolean ig
          }
       }
 
-      QueueConfigurationUtils.applyDynamicQueueDefaults(queueConfiguration, addressSettingsRepository.getMatch(getRuntimeTempQueueNamespace(queueConfiguration.isTemporary()) + queueConfiguration.getAddress().toString()));
+      final AddressSettings addressSettings = addressSettingsRepository.getMatch(getRuntimeTempQueueNamespace(queueConfiguration.isTemporary()) + queueConfiguration.getAddress().toString());
+      QueueConfigurationUtils.applyDynamicQueueDefaults(queueConfiguration, addressSettings);
+
+      if (ActiveMQServerLogger.LOGGER.isEnabled(Logger.Level.WARN)) {
+         if (new AddressImpl(queueConfiguration.getAddress(), configuration.getWildcardConfiguration()).containsWildCard() && addressSettings.getPageStoreName() == null) {
```

Comment:
isn't there a better way instead of new AddressImpl() ?


that's the only concern I have.

Reply:
I share that concern, i did not find a way to get the AddressImpl back from the binding created by the wildcard address manager. It needs some surgery to access.

The addressImpl has a reference to the wildcard config, I guess I could duplicate that logic in place of the new ..

Reply:
@gtully I've sent a Pull Request towards your personal fork:

https://github.com/gtully/activemq-artemis/pull/1


if you can squash it with your commit.  I was looking through the code it was easier to just write the change instead of explaining it.

Reply:
actually, I will merge with my change squashed...

you can follow up with additional commits if you don't agree with anything.


thanks a lot for this PR.. very nice!

Reply:
Thank you, that is a nice improvement 😷